### DISCUSSION TITLE
Angular support

### DISCUSSION THREAD

**Discussion by dedalo** [mikaelvesavuori/figmagic#192] [Ideas]
Hello Figmagic team and community,

First off, thank you for the great work on this tool; 

I've noticed that there is support for generating code tailored for React.I wanted to initiate a discussion around the possibility of adding support for Angular projects. 

1. **Is there any ongoing work or plans in the roadmap to incorporate Angular support?**
2. **If not, are there any technical challenges or concerns that the community should be aware of, which might be preventing this addition?**
3. **Would contributions from the community towards this feature be welcomed?**

I believe having Angular support could significantly widen Figmagic's user base and make it a go-to choice for many Angular developers looking to integrate design tokens and assets from Figma directly into their projects.

Thank you!

**Comment by mikaelvesavuori** [comment]
Hi @dedalo!

Great questions. Thanks for asking!

First and foremost, the React "element generation" support has always been mostly experimental. It's hard to get it right (though we are now seeing more tools out there, like Anima and Amplify Studio, getting it kind of "right") and the current solution supports it in a fairly hacky way. The intent of it was to allow a foundational type of creation, say "boilerplating", and getting documents in order with _pretty rightish_ content - such files would in many cases have to be manually reworked to some extent, anyway.

With that said, to your questions:

1. No.
2. There would probably be no showstoppers (as far as I am aware of), in implementing Angular support. What happens now is roughly that you pass in options (CLI or configuration; you'll need to add options for Angular here; this is the main class for that https://github.com/mikaelvesavuori/figmagic/blob/main/bin/entities/Config/index.ts); the options are passed into 3 high-level "use cases" (where element generation is one https://github.com/mikaelvesavuori/figmagic/blob/main/bin/usecases/createElements.ts; it uses this class to do most of the work https://github.com/mikaelvesavuori/figmagic/blob/main/bin/entities/FigmagicElement/index.ts); and finally at the end, just before writing files to disk, it replaces values in the appropriate templates (see https://github.com/mikaelvesavuori/figmagic/tree/main/templates; much of the work happens in https://github.com/mikaelvesavuori/figmagic/blob/main/bin/frameworks/filesystem/prepFile.ts). You'd need to add the Angular template(s) with the same placeholder values and then make the functional changes to that last file. So it's surely a number of steps (plus following the general contribution standards, adding tests, updating docs...), but it should be doable. The risk factor is totally around whether or not this "template updating" way of doing things would work. Also, there is no good (it's pretty ugly today) way of switching between _completely_ different sets of needs/requirements in terms of files generated etc. All in all: If it can work within the given constraints it's doable or otherwise it will require some under-the-hood refactoring and rethinking.
3. Yes!

**Reply by mikaelvesavuori** [reply]
Just to clarify: Above I was assuming Angular _components_ support. Tokens, however, should be functional already, I should assume (given the options of regular TS/JS/MJS, JSON, CSS variables...). Or is this not working? These options seem entirely platform/framework-agnostic to me.

**Reply by dedalo** [reply]
Yes, the idea is to get the same functionality as react, just a boilerplate of components. the rest is working ok.
